plugins/entityreference/behavior/OgBehaviorHandler.class.php

```diff
@@ -122,11 +122,7 @@ public function OgMembershipCrud($entity_type, $entity, $field, $instance, $lang
     foreach ($items as $item) {
       $gid = $item['target_id'];
 
-      // Must provide correct state in the event that approval is required.
-      if (empty($item['state']) && $entity_type == 'user' && !og_user_access($group_type, $gid, 'subscribe without approval', $entity) && !og_user_access($group_type, $gid, 'administer group')) {
```

Comment:
So I used the same environment as we use in Travis, same PHP built-in webserver, same setup, and I was on this branch, not the main one.
OG Access configured:
![perm-table](https://cloud.githubusercontent.com/assets/114076/26189234/c977a10a-3ba3-11e7-8936-4384294fa42c.png)
Non-member regular Drupal user wants to join:
![join](https://cloud.githubusercontent.com/assets/114076/26189245/d9810974-3ba3-11e7-9cee-53896a5f7472.png)
After join, he sees this:
![unsub](https://cloud.githubusercontent.com/assets/114076/26189255/e6647a36-3ba3-11e7-8124-e039062bba54.png)
And in the database, the state is:
![state-2](https://cloud.githubusercontent.com/assets/114076/26189260/edee2e8c-3ba3-11e7-8023-f3fa38ece074.png)
And state 2 is Pending: https://github.com/Gizra/og/blob/7.x-2.x/og.module#L26 , so it still works as expected.